esmvalcore/config/_config_object.py

```diff
@@ -359,6 +474,142 @@ def _fixed_file_dir(self):
         return self.session_dir / self._relative_fixed_file_dir
 
 
+def _get_user_config_dir_from_cli() -> None | str:
+    """Try to get user-defined configuration directory from CLI arguments.
+
+    The hack of directly parsing the CLI arguments here (instead of using the
+    `fire` or `argparser` module) ensures that the correct user configuration
+    file is used. This will always work, regardless of when this module has
+    been imported in the code.
+
+    Note
+    ----
+    If not called within the `esmvaltool` program, always return `None`.
+
+    """
+    if Path(sys.argv[0]).name != 'esmvaltool':
+        return None
+
+    for arg in sys.argv:
+        for opt in ('--config-dir', '--config_dir'):
+            if opt in arg:
+                # Parse '--config-dir=/dir' or '--config_dir=/dir'
+                partition = arg.partition('=')
+                if partition[2]:
+                    return partition[2]
+
+                # Parse '--config-dir /dir' or '--config_dir /dir'
+                config_idx = sys.argv.index(opt)
+                if config_idx == len(sys.argv) - 1:  # no dir given
+                    return None
+                return sys.argv[config_idx + 1]
+
+    return None
+
+
+def _get_user_config(ignore_internal_env: bool = False) -> tuple[str, Path]:
+    """Get user configuration directory.
+
+    The following directories are considered (descending priority):
+
+    1. Internal `_ESMVALTOOL_USER_CONFIG_DIR_` environment variable (this
```

Comment:
Would it be possible to use the `ESMVALTOOL_CONFIG_DIR` for this instead of adding an extra internal variable? That seems conceptually simpler.

Reply:
The main reason we need the `_ESMVALTOOL_USER_CONFIG_DIR_` variable is that the path given via `--config_dir` is properly given to the diagnostics (these are run with `subprocess`, so `sys.argv` is different for those). Since the path for `--config_dir` and `ESMVALTOOL_CONFIG_DIR` can be different (in the current version but also in the change you suggest above), we need an additional environment variable.

Reply:
Removed in https://github.com/ESMValGroup/ESMValCore/commit/2decd64f0b43f58cb8145c1fd4b1cbd42decf9a6, see comment above